lib/GenomeFileUtil/core/FastaGFFUploaderScript.py

```diff
@@ -0,0 +1,678 @@
+
+        #Provenance has a 1 MB limit.  We may want to add more like the accessions, but to be safe for now not doing that.
+        assembly_provenance = [{"script": __file__, "script_ver": "0.1", "description": "Assembly from FASTA from %s" % (source)}]
+
+        #Report large genomes?
+        hidden=0
+
+        logger.info("Attempting Assembly save for %s" % (assembly["assembly_id"]))
+        assembly_not_saved = True
+        while assembly_not_saved:
+            try:
+                assembly_info =  ws_client.save_objects({"workspace":workspace_name,
+                                                           "objects":[ { "type": "KBaseGenomeAnnotations.Assembly",
+                                                                         "data": assembly,
+                                                                         "name": assembly["assembly_id"],
+                                                                         "hidden" : hidden,
+                                                                         "provenance": assembly_provenance}]}) 
+                assembly_not_saved = False 
+                logger.info("Assembly saved for %s" % (assembly["name"]))
+                assembly_ref = workspace_name+"/"+assembly["assembly_id"]
+            except WorkspaceError as e:
+                logger.info('Logging workspace error on save_objects: {}\n{}'.format(e.message, e.data))
+                raise
+
+    genome = dict()
+    genome["id"]=core_genome_name
+    genome["scientific_name"]=scientific_name
+    genome["assembly_ref"]=assembly_ref
+    genome["features"]=genome_features_list
+    genome["cdss"]=genome_cdss_list
+    genome["mrnas"]=genome_mrnas_list
+    genome["source"]=source
+    genome["domain"]="Eukaryota"
+    genome["genetic_code"]=1
+    genome["gc_content"]=assembly["gc_content"]
+    genome["dna_size"]=assembly["dna_size"]
+
+    if taxon_reference is not None:
+        genome["taxon_ref"] = taxon_reference
+        genome["taxonomy"]=taxonomy
+
+    UserMeta = dict()
+    UserMeta['Taxonomy']=taxonomy
+    UserMeta['Source']=source
+    UserMeta['Domain']="Eukaryota"
+    UserMeta['Source ID']=core_genome_name
+    UserMeta['Name']=scientific_name
+    UserMeta['Genetic code']=1;
+
+    UserMeta['GC content']=assembly["gc_content"]
+    UserMeta['Size']=assembly["dna_size"]
+    UserMeta['Number contigs']=assembly['num_contigs']
+
+    #id_source_version_array = core_genome_name.split("_")
+    #version = "_".join(id_source_version_array[2:])
+    #UserMeta['Version']=version
+
+    UserMeta['url']='https://transfer.api.globusonline.org/v0.10/endpoint/78312867-6d04-11e5-ba46-22000b92c6ec/ls?path=By_Organism_Name//P/PhytozomeV11/';
+
+    if(gff_handle_ref == None):
+        #Upload GFF to shock
+        token = os.environ.get('KB_AUTH_TOKEN') 
+
+        shock_info = script_utils.upload_file_to_shock(logger, shock_service_url, input_gff_file, token=token)
```

Comment:
You should use the DataFileUtils function for uploading to shock, not the old script_utils from the Transform anymore.  We're trying to move away from this dependency.